test/redis_test.rb

```diff
@@ -243,6 +245,49 @@ def test_query_instrumentation
     Semian.unsubscribe(subscriber)
   end
 
+  def test_changes_timeout_when_half_open_and_configured
+    client = connect_to_redis!(half_open_resource_timeout: 0.1)
+
+    Timecop.freeze do
+      @proxy.downstream(:latency, latency: redis_timeout_ms + 200).apply do
+        ERROR_THRESHOLD.times do
+          assert_raises ::Redis::TimeoutError do
+            client.get('foo')
+          end
+        end
+      end
+
+      assert_raises ::Redis::CircuitOpenError do
+        client.get('foo')
+      end
+    end
+
+    time_circuit_half_open = Time.now + ERROR_TIMEOUT + 1
+    Timecop.freeze(time_circuit_half_open) do
+      @proxy.downstream(:latency, latency: (SEMIAN_OPTIONS[:half_open_resource_timeout] * 1000).to_i).apply do
```

Comment:
You could change this latency to 'infinite' and this test would still pass. You'll want to do an `assert_in_delta` to do what you're trying to do here.

Reply:
Right, but if Redis timeouts out after `half_open_resource_timeout`, then we know the shorter timeout was successfully applied.

If the shorter timeout wasn't successfully applied, this test would fail which means it is doing its job.

Reply:
Ah yeah, you're right about that. 👍

Reply:
Added a timing check anyway with `Benchmark`